src/hotspot/share/runtime/javaCalls.cpp

```diff
@@ -357,14 +357,6 @@ void JavaCalls::call_helper(JavaValue* result, const methodHandle& method, JavaC
 
   CompilationPolicy::compile_if_required(method, CHECK);
```

Comment:
I'd like to understand what is going to happen at the line 358. Are we going to compile method with -Xcomp even though the `interp_only` mode is required? If so, do we waist cycles in doing this?
This question is not to Leonid. Maybe @dean-long can answer this?

Reply:
Yes, I believe we will compile the method even in interp-only mode.  As -Xcomp is a mode to stress the compiler, I don't think it matters if we compile eagerly here or wait until interp-only mode is turned off.  Other threads not in interp-only mode will compile the method anyway.  But if it makes sense to change this behavior, then let's do that in a separate bug/RFE.

Reply:
Thank you, Dean.
The issue I see is that the tests are extremely slow with the -Xcomp option.
So, it would be good to avoid unneeded proactive compilations.
Let's address it in a separate bug/RFE. I'm expecting more places where we do the same or similar compilations.